File: Ex3.rb
# Функция для переворачивания числа
def reverse_number(num)
    result = 0
    until num == 0
        digit = num % 10
        if result != 0 || digit != 0
            result = result * 10 + digit
        end
        num /= 10
    end
    result
end

# Функция для переворачивания каждого числа в массиве и вывода без ведущих нулей
def reverse_and_print_numbers(numbers)
    numbers.each do |num|
        puts reverse_number(num)
    end
end

puts "Введите количество чисел:"
n = gets.chomp.to_i # Запрос количества чисел
numbers = []
n.times do |i|
    puts "Введите число ##{i + 1}:"
    numbers << gets.chomp.to_i # Ввод каждого числа
end
reverse_and_print_numbers(numbers) # Переворачиваем и выводим числа без ведущих нулей

File: Ex3.cs
using System;
using System.Collections.Generic;

class Program
{
    static int ReverseNumber(int num)
    {
        int result = 0;
        while (num != 0)
        {
            int digit = num % 10;
            if (result != 0 || digit != 0)
            {
                result = result * 10 + digit;
            }
            num /= 10;
        }
        return result;
    }

    static void ReverseAndPrintNumbers(int n, List<int> numbers)
    {
        for (int i = 0; i < n; i++)
        {
            Console.WriteLine(ReverseNumber(numbers[i]));
        }
    }

    static void Main()
    {
        Console.WriteLine("Введите количество чисел:");
        int n = int.Parse(Console.ReadLine());
        List<int> numbers = new List<int>();
        for (int i = 0; i < n; i++)
        {
            Console.WriteLine($"Введите число #{i + 1}:");
            numbers.Add(int.Parse(Console.ReadLine()));
        }
        ReverseAndPrintNumbers(n, numbers);
    }
}